sdk/data/azcosmos/README.md

```diff
@@ -0,0 +1,202 @@
+
+#### Install the package
+* Install the Azure Cosmos DB SDK for Go with `go get`:
+  ```bash
+  go get -u github.com/Azure/azure-sdk-for-go/tree/feature/cosmos/sdk/data/azcosmos
+  ```
+  
+#### Authenticate the client
+
+In order to interact with the Azure CosmosDB service you'll need to create an instance of the Cosmos Client class. To make this possible you will need an URL and key of the Azure CosmosDB service.
+
+#### Create CosmosClient
+```go
+endpoint := "someEndpoint"
+key := "someKey"
+cred, _ := NewSharedKeyCredential(key)
+    client, err := NewCosmosClient(endpoint, cred, nil)
+    if err != nil {
+        t.Fatalf("Failed to create client: %v", err)
+    }
+
+    return client
+```
+
+## Examples
+
+The following section provides several code snippets covering some of the most common CosmosDB SQL API tasks, including:
+* [Create Cosmos Client](#create-cosmos-client "Create Cosmos Client")
+* [Create Database](#create-database "Create Database")
+* [Create Container](#create-container "Create Container")
+* [CRUD operation on Items](#crud-operation-on-items "CRUD operation on Items")
+
+### Create Cosmos Client
+```go
+// Create a new CosmosClient via the CosmosClientBuilder
+// It only requires endpoint and key, but other useful settings are available
+
+const (
+    CosmosDbEndpoint = "someEndpoint"
+    CosmoDbKey = "someKey"
+)
+
+cred, _ := NewSharedKeyCredential(CosmosDbEndpoint)
+    client := newCosmosClientConnection(CosmoDbKey, cred, &CosmosClientOptions{})
+    if connection == nil {
+        t.Error("Expected connection to be not nil")
+    }
+```
+
+### Create Database
+Using the client created in previous example, you can create a database like this:
+
+```go
+// Get a reference to the container
+// This will create (or read) a database and its container.
+    database := CosmosDatabaseProperties{Id: dbName}
+    resp, err := client.CreateDatabase(ctx, database, nil, nil)
+    if err != nil {
+        t.Fatalf("Failed to create database: %v", err)
+    }
+```
+
+### Create Container
+Using the above created database for creating a container, like this:
+
+```go
+resp, err := database.CreateContainer(context, properties, throughput, nil)
+    if err != nil {
+        t.Fatalf("Failed to create container: %v", err)
+    }
+container := resp.ContainerProperties.Container
+```
+### CRUD operation on Items
+
+```go
+
+// Create an item
+database := client.createDatabase(t, context, client, "itemCRUD")
+    properties := CosmosContainerProperties{
+        Id: "aContainer",
+        PartitionKeyDefinition: PartitionKeyDefinition{
+            Paths: []string{"/id"},
+        },
+    }
+
+    resp, err := database.CreateContainer(context, properties, nil, nil)
+    if err != nil {
+        t.Fatalf("Failed to create container: %v", err)
+    }
+
+    item := map[string]string{
+        "id":    "1",
+        "value": "2",
+    }
+
+    container := resp.ContainerProperties.Container
+    pk, err := NewPartitionKey("1")
```

Comment:
You can just how how to create the PartitionKey like you are doing, and the call to CreateItem